src/Microsoft.Identity.Client/IConfidentialClientApplication.cs

```diff
@@ -40,35 +40,70 @@ namespace Microsoft.Identity.Client
 
         /// <summary>
-        /// TODO(migration): fill in jm's documentation
+        /// Acquires an access token for this application (usually a Web API) from the authority configured in the application,
+        /// in order to access another downstream protected Web API on behalf of a user using the OAuth 2.0 On-Behalf-Of flow.
+        /// (See https://aka.ms/msal-net-on-behalf-of).
+        /// This confidential client application was itself called with a token which will be provided in the
+        /// <paramref name="userAssertion">userAssertion</paramref> parameter.
         /// </summary>
-        /// <param name="scopes"></param>
-        /// <param name="userAssertion"></param>
-        /// <returns></returns>
+        /// <param name="scopes">Scopes requested to access a protected API</param>
+        /// <param name="userAssertion">Instance of <see cref="UserAssertion"/> containing credential information about
+        /// the user on behalf of whom to get a token.</param>
+        /// <returns>A builder enabling you to add optional parameters before executing the token request</returns>
+        /// <remarks>You can also chain the following optional parameters:
+        /// <see cref="AbstractAcquireTokenParameterBuilder{T}.WithExtraQueryParameters(Dictionary{string, string})"/>
+        /// </remarks>
         AcquireTokenOnBehalfOfParameterBuilder AcquireTokenOnBehalfOf(IEnumerable<string> scopes, UserAssertion userAssertion);
 
         /// <summary>
-        /// TODO(migration): fill in jm's documentation
+        /// Computes the URL of the authorization request letting the user sign-in and consent to the application accessing specific scopes in
+        /// the user's name. The URL targets the /authorize endpoint of the authority configured in the application.
+        /// This override enables you to specify a login hint and extra query parameter.
         /// </summary>
-        /// <param name="scopes"></param>
-        /// <returns></returns>
+        /// <param name="scopes">Scopes requested to access a protected API</param>
+        /// <returns>A builder enabling you to add optional parameters before executing the token request to get the
+        /// URL of the STS authorization endpoint parametrized with the parameters</returns>
+        /// <remarks>You can also chain the following optional parameters:
+        /// <see cref="GetAuthorizationRequestUrlParameterBuilder.WithRedirectUri(string)"/>
+        /// <see cref="GetAuthorizationRequestUrlParameterBuilder.WithLoginHint(string)"/>
+        /// <see cref="AbstractAcquireTokenParameterBuilder{T}.WithExtraQueryParameters(Dictionary{string, string})"/>
+        /// <see cref="GetAuthorizationRequestUrlParameterBuilder.WithExtraScopesToConsent(IEnumerable{string})"/>
+        /// </remarks>
         GetAuthorizationRequestUrlParameterBuilder GetAuthorizationRequestUrl(IEnumerable<string> scopes);
```

Comment:
Do we want to move this method to the BaseClientApplication / IBaseClientApplication?, with of course override for PCA and CCA as some query parameters will be different. (eg clientSecret).